src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -373,19 +374,13 @@ public static void WriteAllBytes(string path, byte[] bytes)
             if (bytes == null)
                 throw new ArgumentNullException(nameof(bytes));
 
-            InternalWriteAllBytes(path, bytes);
-        }
-
-        private static void InternalWriteAllBytes(string path, byte[] bytes)
-        {
-            Debug.Assert(path != null);
-            Debug.Assert(path.Length != 0);
-            Debug.Assert(bytes != null);
-
-            using (FileStream fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.Read))
-            {
-                fs.Write(bytes, 0, bytes.Length);
-            }
+#if MS_IO_REDIST
+            using FileStream fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.Read);
+            fs.Write(bytes, 0, bytes.Length);
+#else
+            using SafeFileHandle sfh = OpenHandle(path, FileMode.Create, FileAccess.Write, FileShare.Read);
+            RandomAccess.Write(sfh, bytes, 0);
```

Comment:
`Write` is the public method with a check for `CanSeek`:

https://github.com/dotnet/runtime/blob/ab64ce5f71479549222c99da3a15fd56f8328fe4/src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.cs#L146-L150

https://github.com/dotnet/runtime/blob/ab64ce5f71479549222c99da3a15fd56f8328fe4/src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.cs#L243-L251

to achieve the lack of throwing you need to call the internal `WriteAtOffset` that does not perform the check:

```suggestion
            RandomAccess.WriteAtOffset(sfh, bytes, 0);
```

Reply:
Maybe we need a NonRandomAccess ;-)

In all seriousness,  though,  maybe we should reconsider the type name,  and in this or a subsequent release add overloads that don't take a file offset, use the file's current position,  and thus don't require seeking.

Reply:
Naming is hard ;)

> In all seriousness, though, maybe we should reconsider the type name in this release, and in this or a subsequent release add overloads that don't take a file offset, use the file's current position, and thus don't require seeking.

We could also allow for negative offsets that would mean "read next `n` bytes, ignore the offset". To be honest I don't have a perfect solution for it.